### DISCUSSION TITLE
Problem after update to 0.13.4

### DISCUSSION THREAD

**Discussion by honzamuzzy** [amidaware/tacticalrmm#1140] [Q&A] [answered]
Hi, i have problem after update. Web show 500 Internal Server Error. I try to run update.sh with force parametr and I found that the compilation of the UI would not be completed (see appendix).

Any idea what the problem might be?
Thanks
![picture](https://user-images.githubusercontent.com/93223152/168643405-3dcb589f-3f5f-4afe-adb4-a75d318282e3.png)

**Comment by dinger1986** [comment]
whats your contents of /rmm?

Also can you check any of those services if they are in fact running?

**Comment by wh1te909** [comment, marked as answer]
you need more RAM. increase RAM or add swap to allow for `npm run build` to complete (lately it's using a lot of ram).

**Reply by honzamuzzy** [reply]
Adding RAM works. Thanks

**Reply by bbrendon** [reply]
how much ram? curious.

**Reply by honzamuzzy** [reply]
I originally bought a VPS with 1GB of RAM for testing, so I paid extra for 2GB of RAM

**Reply by wh1te909** [reply]
we figured out the cause, since we enabled typescript support last update, when compiling the frontend js code it uses way more ram. this is only needed though during an update, not at runtime. so for future releases we're looking at compiling and releasing static assets using github actions to offload it instead of you having to compile it on your own vm

**Comment by NiceGuyIT** [comment]
Note: You can limit node's memory usage during compiling with the [`--max-old-space-size`](https://nodejs.org/api/cli.html#--max-old-space-sizesize-in-megabytes) option. It's a tradeoff between garbage collection, available memory, swapping and time. As such, it's specific to the host you're on. I've used this on a host with 1G to compile a React app.

**Reply by wh1te909** [reply]
Yea, I did play around with this, unfortunately setting `--max-old-space-size` to a low value caused compilation to fail with OOM error. Lowest number I could get it to compile successfully was with 768 which still might be too high so we're going to offload it
